java/src/main/java/com/logicalclocks/hsfs/StorageConnector.java

```diff
@@ -379,4 +384,106 @@ public String getPath(String subPath) {
       return null;
     }
   }
+
+  public static class KafkaConnector extends StorageConnector {
+
+    public static final String sparkFormat = "kafka";
+
+    @Getter @Setter
+    private String bootstrapServers;
+
+    @Getter @Setter
+    private SecurityProtocol securityProtocol;
+
+    @Getter
+    private String sslTruststoreLocation;
+
+    @Getter @Setter
+    private String sslTruststorePassword;
+
+    @Getter
+    private String sslKeystoreLocation;
+
+    @Getter @Setter
+    private String sslKeystorePassword;
+
+    @Getter @Setter
+    private String sslKeyPassword;
+
+    @Getter @Setter
+    private SslEndpointIdentificationAlgorithm sslEndpointIdentificationAlgorithm;
+
+    @Getter @Setter
+    private List<Option> options;
+
+    public void setSslTruststoreLocation(String sslTruststoreLocation) {
+      this.sslTruststoreLocation = SparkEngine.getInstance().addFile(sslTruststoreLocation);
+    }
+
+    public void setSslKeystoreLocation(String sslKeystoreLocation) {
+      this.sslKeystoreLocation = SparkEngine.getInstance().addFile(sslKeystoreLocation);
+    }
+
+    public Map<String, String> sparkOptions() {
+      Map<String, String> options = new HashMap<>();
+      options.put(Constants.KAFKA_BOOTSTRAP_SERVERS, bootstrapServers);
+      options.put(Constants.KAFKA_SECURITY_PROTOCOL, securityProtocol.toString());
+      if (!Strings.isNullOrEmpty(sslTruststoreLocation)) {
+        options.put(Constants.KAFKA_SSL_TRUSTSTORE_LOCATION, sslTruststoreLocation);
+      }
+      if (!Strings.isNullOrEmpty(sslTruststorePassword)) {
+        options.put(Constants.KAFKA_SSL_TRUSTSTORE_PASSWORD, sslTruststorePassword);
+      }
+      if (!Strings.isNullOrEmpty(sslKeystoreLocation)) {
+        options.put(Constants.KAFKA_SSL_KEYSTORE_LOCATION, sslKeystoreLocation);
+      }
+      if (!Strings.isNullOrEmpty(sslKeystorePassword)) {
+        options.put(Constants.KAFKA_SSL_KEYSTORE_PASSWORD, sslKeystorePassword);
+      }
+      if (!Strings.isNullOrEmpty(sslKeyPassword)) {
+        options.put(Constants.KAFKA_SSL_KEY_PASSWORD, sslKeyPassword);
+      }
+      // can be empty string
+      if (sslEndpointIdentificationAlgorithm != null) {
+        options.put(
+            Constants.KAFKA_SSL_ENDPOINT_IDENTIFICATION_ALGORITHM, sslEndpointIdentificationAlgorithm.getValue());
+      }
+      if (this.options != null && !this.options.isEmpty()) {
+        Map<String, String> argOptions = this.options.stream()
+            .collect(Collectors.toMap(Option::getName, Option::getValue));
+        options.putAll(argOptions);
+      }
+      return options;
+    }
+
+    @JsonIgnore
+    public String getPath(String subPath) {
+      return null;
+    }
+
+    public Dataset<Row> read(String query, String dataFormat, Map<String, String> options, String path) {
+      throw new NotSupportedException("Reading a Kafka Stream into a static Spark Dataframe is not supported.");
```

Comment:
Why we do not support this? Spark can write kafka source to static DF.
https://spark.apache.org/docs/latest/structured-streaming-kafka-integration.html#creating-a-kafka-source-for-batch-queries

Reply:
Mostly cause I thought it will not be useful. Is it common to read a snapshot of a kafka topic, and derive features from it?

Reply:
I can think of features for example "count_last_hour", "count_last_5_mins". Can it be derived from streaming df as well?